packages/clerk-js/src/utils/syncDigest.ts

```diff
@@ -0,0 +1,26 @@
+import { SHA1 } from './sha1';
+
+type HashValueProps = {
+  sessionId: string | undefined | null;
+  userId: string | undefined | null;
+  orgId: string | undefined | null;
+};
+
+export const syncDigest = (values: HashValueProps) => {
+  const sortedValues = Object.keys(values)
+    .sort()
+    .reduce((accumulator, key) => {
+      if (values[key as keyof HashValueProps]) {
+        // @ts-ignore
+        accumulator.push(values[key]);
+      }
+
+      return accumulator;
+    }, []);
```

Comment:
Can we simplify this using a filter function as discussed?

Reply:
@nikosdouvlis doing with filter returns an array of arrays that we have to map as well. Is this better than reduce?
e.g. 
```  
something.sort()
    .filter(s => values[s])
    .map(s => values[s]);
```

Reply:
THis is very minor so don't worry about it, personally I'd go with `Object.entries` as you don't need to reference `values` outside of the initial entries call :)

For the nested array, you can take a look at https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat